pkg/util/array.go

```diff
@@ -25,6 +26,11 @@ func InArray(needle string, haystack []string) bool {
 	for _, item := range haystack {
```

Comment:
please implements new function `InArrayWithRegexp` instead of modify `InArray` function
`InArray` used in multiple code places, we need to change behavior only for watchNamespaces usecase

Reply:
OK, because of the PR review, I thought it was going to be a "replace". I also think it makes sense to implement separate functions.